File: OOP/Lab 1/java/src/Piece.java
import javax.swing.plaf.TreeUI;

public class Piece {
    private String color;
    private String type;
    private int x;
    private int y;
    private boolean canPromote;

    public Piece(String color, String type, int x, int y){
        this.color=color;
        this.type=type;
        this.x=x;
        this.y=y;
        if (this.type=="PAWN"){
            canPromote=true;
        } else canPromote=false;

    }

    public void promotion(){
        if (y==7 && canPromote== true){
            type="QUEEN";
            canPromote=false;
        }
    }
    public void getPiece(){
        System.out.println(String.format("I am a %S %S",color,type));
    }
    public void getCoords(){
        System.out.println(String.format("%d,%d",x,y));
    }

    public void hasPromotionRights(){
        System.out.println((canPromote==true)?"yes":"no");

    }
}

File: OOP/Lab 1/python/Piece.py
class Piece:
    def __init__(self, color, type, x, y):
        self.color = color
        self.type = type
        self.x = x
        self.y = y
        self.can_promote = True if self.type == "PAWN" else False

    def promotion(self):
        if self.y == 7 and self.can_promote:
            self.type = "QUEEN"
            self.can_promote = False

    def get_piece(self):
        print(f"I am a {self.color} {self.type}")

    def get_coords(self):
        print(f"{self.x},{self.y}")

    def has_promotion_rights(self):
        print("yes" if self.can_promote else "no")
